### PR TITLE
[google_maps] fix markersToChange to actually change modified markers

### PR DESCRIPTION
**Pull request by KlausJokinen**
## Description

*Fixes Marker updating method.
now markers needed to change are actually those which need to be changed*

## Related Issues

*flutter/flutter#27823*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by KlausJokinen** [comment]
I signed it!

**Comment by bogdannedelcu** [comment]
Hello @amirh, can you update us on the status on this one please? Thank you.

**Comment by KlausJokinen** [comment]
@amirh What is the status of this PR?

**Comment by kjeremy** [comment]
Why is this taking so long? This is a pretty important performance fix.

**Comment by KlausJokinen** [comment]
Is there something I can do to speed up processing of the PR?

**Comment by amirh** [comment]
@iskakaushik

File: packages/google_maps_flutter/lib/src/marker.dart [hunk 1]
```diff
@@ -274,15 +274,39 @@ class Marker {
...
+          infoWindow == other.infoWindow &&
+          position == other.position &&
+          rotation == other.rotation &&
+          visible == other.visible &&
+          zIndex == other.zIndex &&
+          onTap == other.onTap;
 
   @override
-  int get hashCode => markerId.hashCode;
+  int get hashCode =>
```
Comment (amirh) [inline packages/google_maps_flutter/lib/src/marker.dart]: nit: [hashValues](https://master-api.flutter.dev/flutter/dart-ui/hashValues.html) is typically used in the Flutter codebase.

File: packages/google_maps_flutter/lib/src/marker_updates.dart [hunk 2]
```diff
@@ -36,10 +40,8 @@ class _MarkerUpdates {
         .map(idToCurrentMarker)
         .toSet();
 
-    final Set<Marker> _markersToChange = currentMarkerIds
-        .intersection(prevMarkerIds)
-        .map(idToCurrentMarker)
-        .toSet();
+    final Set<Marker> _markersToChange =
+        currentMarkersSet.difference(previousMarkersSet);
```
Comment (amirh) [inline packages/google_maps_flutter/lib/src/marker_updates.dart]: Won't this include newly added markers as well?

**Comment by amirh** [comment]
Left a few comments, Note that multiple tests and presubmits are failing, they will need to be fixed before a PR can be merged.

Generally speaking we are prioritizing issues by the number of thumbs up, so the best way to get it prioritize is to thumb up flutter/flutter#27823. See the [review process section](https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#the-review-process).

File: packages/google_maps_flutter/lib/src/marker.dart (hunk 1 above)
Comment (KlausJokinen) [reply packages/google_maps_flutter/lib/src/marker.dart]: What do you mean by this?

Comment (amirh) [reply packages/google_maps_flutter/lib/src/marker.dart]: e.g https://github.com/flutter/plugins/blob/80318be69aa81b6b3b089285a6d52cb588c8f98e/packages/google_maps_flutter/lib/src/location.dart#L49

**Comment by KlausJokinen** [comment]
@amirh I've now changed code accordingly to above comments. Thanks for the tip regarding `_markersToChange`. There is a noticeable performance improvement when changing lot of markers at once.

**Comment by bogdannedelcu** [comment]
@amirh @iskakaushik  can you check if this is ok? We need google map to be as fast as possible. Thanks

**Comment by KlausJokinen** [comment]
Fixed in https://github.com/flutter/plugins/pull/1933

**Pull request closed without merging**